relation/src/gadgets/logic.rs

```diff
@@ -192,6 +192,47 @@ impl<F: PrimeField> PlonkCircuit<F> {
         self.insert_gate(&wire_vars, Box::new(CondSelectGate))?;
         Ok(y)
     }
+
+    /// Constrain a variable `y` to equal `f_0` if `b` is zero, or `f_1` if `b` is
+    /// one. Return error if variables are invalid.
```

Comment:
Maybe mention that the argument `b: BoolVar` must be a constrained to be a boolean variable. Because one can use the function `new_unchecked(inner: usize) -> BoolVar` instead of the proper constructor `create_boolean_variable` which adds the boolean contraint.